src/pages/leave/LeaveApplicationDetails.tsx

```diff
@@ -0,0 +1,387 @@
+import React, { useState, useEffect, ChangeEvent } from 'react';
+import '../../styles/common/common.scss';
+import '../../styles/pages/leaveApplicationDetails.scss';
+import { useNavigate } from 'react-router';
+import { useSearchParams } from 'react-router-dom';
+import { LeftOutlined } from '@ant-design/icons';
+import {
+  Button,
+  Card,
+  Collapse,
+  Input,
+  Select,
+  Spin,
+  Tag,
+  Tooltip,
+  Typography
+} from 'antd';
+import { LeaveApplication, LeaveStatus, LeaveType } from 'src/models/types';
+import asyncFetchCallback from 'src/services/util/asyncFetchCallback';
+import {
+  cancelLeaveApplication,
+  editLeaveApplication,
+  getLeaveApplicationById
+} from 'src/services/leaveService';
+import { DatePicker } from 'antd';
+import moment, { Moment } from 'moment';
+import TimeoutAlert, { AlertType } from 'src/components/common/TimeoutAlert';
+import ConfirmationModalButton from 'src/components/common/ConfirmationModalButton';
+
+const { Panel } = Collapse;
+const { RangePicker } = DatePicker;
+const { TextArea } = Input;
+
+const leaveOptions = [
+  {
+    label: 'Annual',
+    value: 'ANNUAL'
+  },
+  { label: 'Childcare', value: 'CHILDCARE' },
+  {
+    label: 'Compassionate',
+    value: 'COMPASSIONATE'
+  },
+  {
+    label: 'Parental',
+    value: 'PARENTAL'
+  },
+  {
+    label: 'Sick',
+    value: 'SICK'
+  },
+  {
+    label: 'Unpaid',
+    value: 'UNPAID'
+  }
+];
+
+const LeaveApplicationDetails = () => {
+  const navigate = useNavigate();
+
+  const [searchParams] = useSearchParams();
+  const id = searchParams.get('id');
+
+  const [originalLeaveApplication, setOriginalLeaveApplication] =
+    useState<LeaveApplication>();
+  const [updatedLeaveApplication, setUpdatedLeaveApplication] =
+    useState<LeaveApplication>();
+  const [selectedDate, setSelectedDate] = useState<[Moment, Moment]>();
+  const [edit, setEdit] = useState<boolean>(false);
+  const [alert, setAlert] = React.useState<AlertType | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setLoading(true);
```

Comment:
put the setLoading into if statement, if not if id is null it will just load perpetually